src/Doctrine/EventListener/PublishMercureUpdatesListener.php

```diff
@@ -190,35 +190,14 @@ private function storeObjectToPublish(object $object, string $property): void
 
         $options['enable_async_update'] ??= true;
 
-        if ($options['topics'] ?? false) {
-            $topics = [];
-            foreach ((array) $options['topics'] as $topic) {
-                if (!\is_string($topic)) {
-                    $topics[] = $topic;
-                    continue;
-                }
-
-                if (!str_starts_with($topic, '@=')) {
-                    $topics[] = $topic;
-                    continue;
-                }
-
-                if (null === $this->expressionLanguage) {
-                    throw new \LogicException('The "@=" expression syntax cannot be used without the Expression Language component. Try running "composer require symfony/expression-language".');
-                }
-
-                $topics[] = $this->expressionLanguage->evaluate(substr($topic, 2), ['object' => $object]);
-            }
-
-            $options['topics'] = $topics;
-        }
-
         if ('deletedObjects' === $property) {
             $types = $operation instanceof HttpOperation ? $operation->getTypes() : null;
             if (null === $types) {
                 $types = [$operation->getShortName()];
             }
 
+            $this->evaluateTopics($options, $object);
```

Comment:
Can you add a comment explaining why topics are evaluated here instead of `publishUpdate`, and not for create/update, please?